loader/src/main/java/net/neoforged/fml/loading/moddiscovery/locators/JarInJarDependencyLocator.java

```diff
@@ -109,9 +109,12 @@ private String formatError(final ModWithVersionRange modWithVersionRange) {
     }
 
     protected String identifyMod(final IModFile modFile) {
-        if (modFile.getModFileInfo() == null || modFile.getModInfos().isEmpty()) {
+        if (modFile.getModFileInfo() == null) {
             return modFile.getFileName();
         }
+        if (modFile.getModInfos().isEmpty()) {
```

Comment:
This could use a comment explaining it's for libraries. It would also be good to explain why the `modFile.getModFileInfo() == null` check above is necessary but that's not related to your PR.